roles/openshift_on_openstack/tasks/main.yml

```diff
@@ -188,50 +201,84 @@
+    # Run the Ansible playbook that checks the prerequisites in OpenStack.
+    - name: Checking the prerequisites for cluster provisioning in OpenStack
+      # Using two inventory flags here as the README instructs.
+      shell: "{{ ansible_playbook }} -vvv --user openshift -i inventory -i {{ inventory_py }} {{ openshift_openstack_dir }}/openshift-cluster/prerequisites.yml 2>&1 >> {{ prerequisite_log }}"
+      args:
+        # Use bash to get the posix style redirects.
+        executable: /bin/bash
+  rescue:
+    # Get some debug information that makes it easier to file OpenShift bugs.
+    - name: Running commands that create debug information needed to open issues
+      shell: "{{ item }}"
+      register: debug_information
+      with_items:
+        - "printf \"$(uname -srm)\n$(cat /etc/os-release)\n\""
+        - "ansible --version"
+        - "cd openshift-ansible ; git describe"
+        - "rpm -q atomic-openshift-utils openshift-ansible"
+      changed_when: false
+      ignore_errors: true
+
+      # Create a string of just the standard output.
+      - name: Create a debug string
+        set_fact:
+          debug_string: "{{ debug_string }}\n{{ item['stdout'] }}"
+        with_items: "{{ debug_information['results'] }}"
+        ignore_errors: true
+
+      # Write the debug information to a file.
+      - name: Write the debug information to a file
+        copy:
+          content: "{{ debug_string }}"
+          dest: "{{ artifacts_directory }}/debug_information.txt"
+        delegate_to: localhost
+        ignore_errors: true
+  always:
+    - name: Copy the log file to the artifacts directory
+      synchronize:
+        src: "{{ prerequisite_log }}"
+        dest: "{{ artifacts_directory }}"
+        mode: pull
+      ignore_errors: true
+
 
 # Set the path for the OpenStack provision log file.
 - name: Creating a variable for the OpenStack provision log file
   set_fact:
     openstack_provision_log: "{{ ansible_user_dir }}/openstack_provision.log"
 
-# Run the Ansible playbook that provisions the OpenStack resources.
-- name: Creating the OpenStack resources
-  # Using two inventory flags here as the README instructs.
-  shell: "{{ ansible_playbook }} -vvv --user openshift -i inventory -i {{ inventory_py }} {{ openshift_openstack_dir }}/openshift-cluster/provision.yml 2>&1 >> {{ openstack_provision_log }}"
-  args:
-    # Use bash to get the posix style redirects.
-    executable: /bin/bash
+- name: OpenStack provision block
+  block:
+    # Run the Ansible playbook that provisions the OpenStack resources.
+    - name: Creating the OpenStack resources
+      # Using two inventory flags here as the README instructs.
+      shell: "{{ ansible_playbook }} -vvv --user openshift -i inventory -i {{ inventory_py }} {{ openshift_openstack_dir }}/openshift-cluster/provision.yml 2>&1 >> {{ openstack_provision_log }}"
+      args:
+        # Use bash to get the posix style redirects.
+        executable: /bin/bash
+  always:
```

Comment:
In the future we can add a few rescue plays which dump information regarding the instances, the networking and the OpenStack health etc to see if we can determine why the overcloud failed to provision the instances.